core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
 
 set -e
 
-echo "< Starting Kubestellar container >-------------------------"
+KUBESTELLAR_SERVICE="kubestellar"
 
-# Try to guess the route/ingress
-if [ -z "$EXTERNAL_HOSTNAME" ]; then
-    if kubectl get route kubestellar-route &> /dev/null; then
-        EXTERNAL_HOSTNAME=$(kubectl get route kubestellar-route -o yaml -o jsonpath={.spec.host} 2> /dev/null)
+
+function echoerr() {
+   echo "ERROR: $1" >&2
+}
+
+
+function ensure_inv() {
+    while IFS=':' read -ra items; do
+        for i in "${items[@]}"; do
+            if [ "$i" == "" ] ; then
+                break
+            fi
+            if ! kubectl ws "$i" &> /dev/null ; then #
+                if ! kubectl ws create "$i" --enter &> /dev/null ; then
+                    echoerr 'unable to create "$i"!'
+                    break
+                fi
+            fi
+        done
+    done <<< "$1"
+}
+
+function ensure_invs() {
+    initial_ws=$(kubectl ws . --short)
+
+    while IFS=',' read -ra items; do
+        for i in "${items[@]}"; do
+            if [ "$i" == "" ] ; then
+                continue
+            fi
+            echo "Ensuring INV: $i"
+            ensure_inv "$i"
+            kubectl ws use $initial_ws &> /dev/null # go back to the initial state
+        done
+    done <<< "$1"
+
+    kubectl ws use $initial_ws &> /dev/null
+}
+
+
+function ensure_wmw() {
```

Comment:
Now that we avoid workspace hierarchy this can be much simpler.

Reply:
I'm not sure I understand